lib/manager/npm/post-update/lerna.ts

```diff
@@ -40,24 +41,21 @@ export async function generateLockFiles(
   const cmd = [];
   let cmdOptions = '';
   try {
+    const toolConstraints: ToolConstraint[] = [];
     if (lernaClient === 'yarn') {
-      let installYarn = 'npm i -g yarn';
-      const yarnCompatibility = config.constraints?.yarn;
-      if (validRange(yarnCompatibility)) {
-        installYarn += `@${quote(yarnCompatibility)}`;
-      }
-      preCommands.push(installYarn);
+      toolConstraints.push({
+        toolName: 'yarn',
+        constraint: config.constraints?.yarn,
+      });
       if (skipInstalls !== false) {
         preCommands.push(getOptimizeCommand());
       }
       cmdOptions = '--ignore-scripts --ignore-engines --ignore-platform';
     } else if (lernaClient === 'npm') {
-      let installNpm = 'npm i -g npm';
-      const npmCompatibility = config.constraints?.npm;
-      if (validRange(npmCompatibility)) {
-        installNpm += `@${quote(npmCompatibility)} || true`;
-      }
-      preCommands.push(installNpm, 'hash -d npm');
```

Comment:
Might still need to do `hash -d npm`

Reply:
yes, maybe i can fix it at buildpack before we merge this